File: src/main/java/bilibili/lingaly/tutorial/item/ModItems.java
package bilibili.lingaly.tutorial.item;

import bilibili.lingaly.tutorial.ModItemGroups;
import bilibili.lingaly.tutorial.Tutorial;
import bilibili.lingaly.tutorial.item.advanced.FoodaTip;
import bilibili.lingaly.tutorial.item.advanced.Probe;
import net.fabricmc.fabric.api.item.v1.FabricItemSettings;
import net.minecraft.item.Item;
import net.minecraft.util.Identifier;
import net.minecraft.util.registry.Registry;

public class ModItems {

    public static Item TUTORIAL_INGOT = registerItem("tutorial_ingot",
            new Item(new FabricItemSettings().group(ModItemGroups.TUTORIAL)));
    public static Item TUTORIAL_NUGGET = registerItem("tutorial_nugget",
            new Item(new FabricItemSettings().group(ModItemGroups.TUTORIAL)));
    public static Item RAW_TUTORIAL = registerItem("raw_tutorial",
            new Item(new FabricItemSettings().group(ModItemGroups.TUTORIAL)));

    public static Item FOODA = registerItem("fooda",
            new FoodaTip(new FabricItemSettings()
                    .group(ModItemGroups.TUTORIAL)
                    .food(ModFoodComponent.fooda)));
    public static Item PROBE = registerItem("probe",
            new Probe(new FabricItemSettings()
                    .group(ModItemGroups.TUTORIAL)));

    public static Item registerItem(String name, Item item){
        return Registry.register(Registry.ITEM , new Identifier(Tutorial.MOD_ID , name), item);
    }

    public static void ModItemRegister(){}

}

File: src/main/kotlin/bilibili/lingaly/ktmccouse/items/ModItems.kt
package bilibili.lingaly.ktmccouse.items

import bilibili.lingaly.ktmccouse.Couse
import bilibili.lingaly.ktmccouse.ModItemGroups
import bilibili.lingaly.ktmccouse.items.advanced.FoodaTip
import bilibili.lingaly.ktmccouse.items.advanced.Probe
import net.fabricmc.fabric.api.item.v1.FabricItemSettings
import net.minecraft.item.Item
import bilibili.lingaly.easyfabricregisterlib.EFRLModRegister

object ModItems {

    private const val MOD_ID = Couse
        .MOD_ID

    val TIN_INGOT: Item = EFRLModRegister
        .registerItem(
        MOD_ID,
        "tin_ingot",
        ModItemGroups
            .TIN
    )

    @Suppress("unused")
    val TIN_NUGGET: Item = EFRLModRegister
        .registerItem(
        MOD_ID,
        "tin_nugget",
        ModItemGroups
            .TIN
    )

    @Suppress("unused")
    val RAW_TIN: Item = EFRLModRegister
        .registerItem(
        MOD_ID,
        "raw_tin",
        ModItemGroups
            .TIN
    )

    @Suppress("unused")
    val FOODA: Item = EFRLModRegister
        .registerItem(
        MOD_ID,
        "fooda",
        FoodaTip(
            FabricItemSettings()
                .group(
                    ModItemGroups
                        .TIN
                )
                .food(
                    ModFoodComponent
                        .fooda
                )
        )
    )

    @Suppress("unused")
    val PROBE: Item = EFRLModRegister
        .registerItem(
            MOD_ID,
            "probe",
            Probe(
                FabricItemSettings()
                    .group(
                        ModItemGroups
                            .TIN
                    )
            )
        )

    fun modItemRegister() {}

}